plugins/inputs/intel_baseband/sock_connector.go

```diff
@@ -0,0 +1,108 @@
+//go:build linux && amd64
+
+package intel_baseband
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+const (
+	// Command code
+	clearLogCmdID   = 0x4
+	deviceDataCmdID = 0x9
+)
+
+type SocketConnector struct {
+	pathToSocket          string
+	accessTimeout         time.Duration
+	waitForTelemetryDelay time.Duration
+	connection            net.Conn
+}
+
+func (sc *SocketConnector) dumpTelemetryToLog() error {
+	// clean the log to have only the latest metrics in the file
+	err := sc.sendCommandToSocket(clearLogCmdID)
+	if err != nil {
+		return fmt.Errorf("failed to send clear log command: %w", err)
+	}
+
+	// fill the file with the latest metrics
+	err = sc.sendCommandToSocket(deviceDataCmdID)
+	if err != nil {
+		return fmt.Errorf("failed to send device data command: %w", err)
+	}
+
+	//time necessary for pf-bb-config to update the data in the file
+	time.Sleep(sc.waitForTelemetryDelay)
```

Comment:
@srebhan Approach was changed a little bit. 

This is tricky, because pf-bb-config has strange approach to clearing telemetry log file and writing new telemetry. 
So now plugin waits a small amount of time after sending clear and write telemetry commands to pf-bb-config and after that starts checking for completeness of log file. 
Log will be completed when all these conditions are met:
- modification time has been changed (in comparison to previous modification time during successful read),
- file is not empty (file can be empty when plugin reads it just after file was cleared but just before new telemetry is saved to this file),
- file doesn't contain clear_log command (this command is written to file just after clear command is sent to pf-bb-config and just before file is actually cleared).

User has now possibility to change `wait_for_telemetry_timeout` value (instead of `wait_for_telemetry_delay`). 
This is duration that defines **maximum** time plugin will wait for pf-bb-config to write telemetry to the log file (defaults to `1s`). Of course if log is completed earlier (as mentioned above), it will be read earlier.